File: frontend/src/types/company.ts
/**
 * 公司领域类型定义
 */

export interface Company {
  id: string
  unifiedSocialCreditCode: string
  name: string
  shortName: string | null
  englishName: string | null
  formerName: string | null
  legalRepresentative: string | null
  chairman: string | null
  manager: string | null
  secretary: string | null
  regCapital: number | null
  setupDate: string | null
  province: string | null
  city: string | null
  regAddress: string | null
  officeAddress: string | null
  website: string | null
  industry: string | null
  mainBusiness: string | null
  businessScope: string | null
  introduction: string | null
  employees: number | null
  controllerName: string | null
  controllerType: string | null
  updatedAt: string
  createdAt: string
}

export interface CompanyDetail extends Company {
  stocks: StockBrief[]
}

export interface StockBrief {
  stockCode: string
  name: string
  market: string | null
  exchange: string | null
  listDate: string | null
}

export interface CreateCompanyRequest {
  unifiedSocialCreditCode: string
  name: string
  shortName?: string
  englishName?: string
  formerName?: string
  legalRepresentative?: string
  chairman?: string
  manager?: string
  secretary?: string
  regCapital?: number
  setupDate?: string
  province?: string
  city?: string
  regAddress?: string
  officeAddress?: string
  website?: string
  industry?: string
  mainBusiness?: string
  businessScope?: string
  introduction?: string
  employees?: number
  controllerName?: string
  controllerType?: string
}

File: backend/src/main/java/org/cwowhappy/securityanalyze/company/domain/model/Company.java
package org.cwowhappy.securityanalyze.company.domain.model;

import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 公司领域实体（Aggregate Root）。
 * 纯 POJO，无框架依赖。
 */
@Getter
@Builder
public class Company {

    private final CompanyId id;
    private String unifiedSocialCreditCode;
    private String name;
    private String shortName;
    private String englishName;
    private String formerName;
    private String legalRepresentative;
    private String chairman;
    private String manager;
    private String secretary;
    private BigDecimal regCapital;
    private LocalDate setupDate;
    private String province;
    private String city;
    private String regAddress;
    private String officeAddress;
    private String website;
    private String industry;
    private String mainBusiness;
    private String businessScope;
    private String introduction;
    private Integer employees;
    private String controllerName;
    private String controllerType;
    private LocalDateTime updatedAt;
    private LocalDateTime createdAt;
}
